xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/internal/skin/ClassLoaderSkinResource.java

```diff
@@ -51,11 +52,23 @@ public class ClassLoaderSkinResource extends AbstractSkinResource
     public ClassLoaderSkinResource(String path, String resourceName, ResourceRepository repository,
         ClassLoader classloader, Provider<XWikiContext> xcontextProvider, URLConfiguration urlConfiguration)
     {
-        super(path, resourceName, repository, xcontextProvider, urlConfiguration);
+        super(createId(classloader, path), path, resourceName, repository, null, xcontextProvider, urlConfiguration);
 
         this.classLoader = classloader;
     }
 
+    private static String createId(ClassLoader classloader, String path)
+    {
+        return "classloader:" + classloader.getName() + ':' + path;
+    }
+
+    @Override
+    public Instant getInstant() throws Exception
+    {
+        // TODO: return the date of the package file holding the resource
```

Comment:
Future improvment or you forgot it?

Reply:
Yes, more a future improvement since it's not really needed yet, but could become a nice to have.